src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -75,7 +75,8 @@ static int DropOrphanedShardsForMove(bool waitForLocks);
 static bool TryDropShardOutsideTransaction(OperationId operationId,
```

Comment:
I cannot see anything related to connection management here but few things to note for excessive connections:

- If you use `FORCE_NEW_CONNECTION` flag, as the name suggests, it'd establish a new connection.
- Instead, if you pass `connectionFlag = 0`, it'd try to use one of the existing connections to the given node, with the same user/database.
- At the end of the transaction, Citus closes all the connections, only keeps 1 (e.g., max_cached_conns_per_worker), see https://github.com/citusdata/citus/blob/3bf3d9c19b8631c6848f0c52865d1dad0e8693c4/src/backend/distributed/connection/connection_management.c#L1472-L1482



With this basic information out there, there are few strategies to avoid excessive connections:
- As @JelteF aimed here, keep the pointer to the connection and re-use: https://github.com/citusdata/citus/pull/6314/files
- Avoid FORCE_NEW_CONNECTION and let connection manager to pick the existing connection for you: https://github.com/citusdata/citus/commit/8be8d9ceb9bd127f9f5a34b3cb52c0e4090ecd42
- We definitely have a similar problem as above on TryDropShardOutsideTransaction. We should try re-use existiing

Reply:
Thanks Onder for the suggestion. The plan was to put the changes as part of the other PR.
I have removed all connection changes now and will continue on other PR.